datafusion/physical-plan/src/aggregates/group_values/group_column.rs

```diff
@@ -63,75 +63,25 @@ pub trait GroupColumn: Send + Sync {
     fn take_n(&mut self, n: usize) -> ArrayRef;
 }
 
-/// An implementation of [`GroupColumn`] for primitive values which are known to have no nulls
-#[derive(Debug)]
-pub struct NonNullPrimitiveGroupValueBuilder<T: ArrowPrimitiveType> {
```

Comment:
the idea is to use a const generic parameter here rather than a second copy of the code